vision/obstacle/obstacle_tracker.py

```diff
@@ -0,0 +1,95 @@
+"""
+Keeps track of detected obstacles between frames
+"""
+
+import cv2
+import numpy as np
+from vision.bounding_box import BoundingBox
+
+
+class Obstacle:
+    """
+    Properties of an obstacle in the frame.
+
+    Parameters
+    ----------
+    bbox: BoundingBox
+        bounding box of obstacle 
+    """
+    def __init__(self, bbox: BoundingBox):
+        self.bounding_box = bbox
+        self.center = np.mean(bbox)
+        self.frames_persisted = 0
+
+
+class ObstacleTracker:
+    """
+    Track obstacles between frames.
+    """
+    def __init__(self):
+        self.MVMT_TOLERANCE = 0.10 # bounding box displacement tolerance per frame
+        self.PERSISTENCE_THRESHOLD = 5 # number of frames for an obstacle to be considered consequential
+        
+        self.obstacles = np.array([]) # buffer that stores current Obstacles in view
+
+    def _isSameObstacle(self, obj1: Obstacle, obj2: Obstacle) -> bool:
+        """
+        Returns True if the position of obj1 is within self.MVMT_TOLERANCE of obj2 
+
+        Parameters
+        ----------
+        obj1: Obstacle
+            The obstacle of reference
+        obj2: Obstacle
+            The obstacle in question
+        """
+        return obj1.center >= (1-self.MVMT_TOLERANCE)*obj2.center and obj1.center <= (1+self.MVMT_TOLERANCE)*obj2.center

```

Comment:
One more thing with the math here: You are multiplying the center value by the movement tolerance, but you want to be multiplying by the total dimensions of the image. Otherwise, obstacles closer to the top left corner will have a smaller tolerance than obstacles in the lower right.

Reply:
Second sentence is misworded, should read:
You are multiplying the center value by the movement tolerance, but you want to be multiplying the total dimensions of the image by the movement tolerance.

Also, the image dimensions are 1920x1080.